pipenv/utils.py

```diff
@@ -535,7 +535,7 @@ def resolve_deps(deps, which, which_pip, project, sources=None, verbose=False, p
     markers_lookup = {}
 
     python_path = which('python', allow_global=allow_global)
-    backup_python_path = shellquote(sys.executable)
+    backup_python_path = sys.executable
```

Comment:
I’m not positive this was added on a whim, this is specifically to handle paths with spaces in them. I’d like to get testing around that area before we remove this.

Reply:
I did some noodling on how to do such testing, but didn't really come up with a great way of doing it. That said, my WORKON_PATH change in #1245 will probably give us a route to producing virtualenvs (and therefore `backup_python_path`s) with paths of our choosing, so I'll have a play around with that.

(I did some local playing around with subprocess (on Linux), and it didn't seem like whitespace (or even quotes) in paths needed to be escaped to work.)

Reply:
@nateprewitt it's going to be really hard to keep writing extra layers of tests to do specific things that trick the resolver into failing to the secondary resolver and such if we are just trying to test this... this call is functionally identical to the line above it if `allow_global=True`, so we can possibly change this code to say:

```python
python_path = which('python', allow_global=allow_global)
backup_python_path = which('python', allow_global=True)
```

And keep the escaping logic (if necessary) in that one place.

The path just gets passed as a string to the hacked python version, it's never used as a shell input (and therefore preserves spaces and doesn't require escaping).  I tested this by hand, but I have no idea how we would confirm that the resolver still works with a python path with spaces unless we can run an appveyor instance with a python path with spaces in it.  In general do you think it's better to shell escape things at all possible points? Or should we try to keep that logic relegated only to the points where we are passing things to the shell?